themes-default/slim/views/vue-components/name-pattern.mako

```diff
@@ -0,0 +1,378 @@
+             * If none provided we asume this is the default episode naming component.
+             * And that means there will be no checkbox available to enable/disable it.
+             */
+            namingCustom: {
+                type: String,
+                default: ''
+            },
+            enabledNamingCustom: {
+                type: Boolean,
+                default: false
+            }
+        },
+        data() {
+            return {
+                namingPattern: '',
+                customName: '',
+                namingPresets: [],
+                showLegend: false,
+                namingExample: '',
+                namingExampleMulti: '',
+                animeNamingType: this.animeType,
+                isEnabled: this.enabledNamingCustom,
+                isMulti: true
+            }
+        },
+        methods: {
+            getDateFormat(format) {
+                return dateFns.format(new Date(), format);
+            },
+            async testNaming(pattern, isMulti, animeType) {
+                console.debug('Test pattern ' + pattern + ' for ' + (isMulti) ? 'multi' : 'single' + ' ep');
+                const params = {
+                    pattern: pattern,
+                    anime_type: animeType
+                }
+
+                if (isMulti) {
+                    params.multi = (isMulti) ? 1 : 0 
+                }
+
+                try {
+                    const result = await apiRoute.get('config/postProcessing/testNaming', {params: params});
+                    return result.data;
+                } catch (e) {
+                    console.warning(e);
+                }
+            },
+            async updatePatternSamples() {
+                // If it's a custom pattern, we need to get the custom pattern from this.customName
+                const namingPattern = (this.isCustom) ? this.customName : this.namingPattern; 
+                
+                // Update single
+                this.namingExample = await this.testNaming(namingPattern, false, this.animeNamingType) + '.ext';
+                console.debug('Result of naming pattern check: ' + this.namingExample);
+                
+                // Update multi if needed
+                if (this.isMulti) {
+                    this.namingExampleMulti = await this.testNaming(namingPattern, this.isMulti, this.animeNamingType) + '.ext';
+                }
+            }
+        },
+        computed: {
+            isCustom() {
+                return !this.namingPresets.includes(this.namingPattern) || this.namingPattern === 'Custom...';
+            },
+            selectedNamingPattern: {
+                get: function() {
+                    return (this.isCustom) ? 'Custom...' : this.namingPattern;
+                },
+                set: function(value) {
+                    this.namingPattern = value;
+                }
+            }
+        },
+        mounted() {
+            this.namingPattern = this.pattern;
```

Comment:
Change this to.

Reply:
You could also add `namingPattern` and `presets` here so you're not setting anything directly to `this.*`.

Reply:
But why? You go from one line to two lines of code. What is the benefit of this?

Reply:
Just the same reason you'd use a const anywhere else.

If you use `this.` you can still set it to another value which is wrong since you should be using the store or local data. If you're using a const there's no issue there and it makes it more clear that you're import `read-only` data.